src/webhook/handlers/processImage.js

```diff
@@ -0,0 +1,223 @@
+        orderBy: [{ _score: DESC }]
+        first: 4
+      ) {
+        edges {
+          score
+          node {
+            id
+            articleType
+            attachmentUrl(variant: THUMBNAIL)
+            attachmentHash
+          }
+        }
+      }
+    }
+  `({
+    mediaUrl: proxyUrl,
+  });
+
+  if (ListArticles.edges.length) {
+    // Track if find similar Articles in DB.
+    visitor.event({ ec: 'UserInput', ea: 'ArticleSearch', el: 'ArticleFound' });
+
+    // Track which Article is searched. And set tracking event as non-interactionHit.
+    ListArticles.edges.forEach(edge => {
+      visitor.event({
+        ec: 'Article',
+        ea: 'Search',
+        el: edge.node.id,
+        ni: true,
+      });
+    });
+
+    const articleOptions = ListArticles.edges
+      .map(({ node: { attachmentUrl, id }, score }, index) => {
+        const imgNumber = index + 1;
+        const displayTextWhenChosen = ngettext(
+          msgid`No.${imgNumber}`,
+          `No.${imgNumber}`,
+          imgNumber
+        );
+
+        const similarity = score - 1;
+        const scoreInPercent = Math.floor(similarity * 100);
+        const similarityEmoji = ['😐', '🙂', '😀', '😃', '😄'][
+          Math.floor(similarity * 4.999)
+        ];
+        const looks = ngettext(
+          msgid`Looks ${scoreInPercent}% similar`,
+          `Looks ${scoreInPercent}% similar`,
+          scoreInPercent
+        );
+
+        //show url attachmentUrl
+        return {
+          type: 'bubble',
+          direction: 'ltr',
+          header: {
+            type: 'box',
+            layout: 'horizontal',
+            spacing: 'md',
+            paddingBottom: 'none',
+            contents: [
+              {
+                type: 'text',
+                text: similarityEmoji,
+                flex: 0,
+              },
+              {
+                type: 'text',
+                text: displayTextWhenChosen + ', ' + looks,
+                gravity: 'center',
+                size: 'sm',
+                weight: 'bold',
+                wrap: true,
+                color: '#AAAAAA',
+              },
+            ],
+          },
+          hero: {
+            type: 'image',
+            url: attachmentUrl,
+            size: 'full',
+          },
+          footer: {
+            type: 'box',
+            layout: 'horizontal',
+            contents: [
+              {
+                type: 'button',
+                action: createPostbackAction(
+                  t`Choose this one`,
+                  id,
+                  t`I choose “${displayTextWhenChosen}”`,
+                  data.sessionId,
+                  'CHOOSING_ARTICLE'
+                ),
+                style: 'primary',
+                color: '#ffb600',
+              },
+            ],
+          },
+        };
+      })
+      .slice(0, 9) /* flex carousel has at most 10 bubbles */;
+
+    // Show "no-article-found" option
+    //
+    articleOptions.push({
```

Comment:
I think we should not show "None of these messages matches mine" if there is a 100% match.

Because if the user chooses this option and tries to submit the image, `MediaManager` will find that the uploaded file has 100% match and does not perform upload anyway.